TYPO3.Flow/Classes/TYPO3/Flow/I18n/TranslationProvider/XliffTranslationProvider.php

```diff
@@ -67,35 +67,65 @@ public function injectPluralsReader(\TYPO3\Flow\I18n\Cldr\Reader\PluralsReader $
     }
 
     /**
-     * Returns translated label of $originalLabel from a file defined by $sourceName.
+     * Returns translated $stringToTranslate from a file defined by $sourceName using the function $functionName of the XliffModel.
      *
      * Chooses particular form of label if available and defined in $pluralForm.
      *
-     * @param string $originalLabel Label used as a key in order to find translation
+     * @param string $functionName The name of the function in the XliffModel to get the translation from
+     * @param string $stringToTranslate String passed to function in order to find translation
      * @param \TYPO3\Flow\I18n\Locale $locale Locale to use
-     * @param string $pluralForm One of RULE constants of PluralsReader
+     * @param float|int|null $quantity A number to find plural form for (float or int), NULL to not use plural forms
      * @param string $sourceName A relative path to the filename with translations (labels' catalog)
      * @param string $packageKey Key of the package containing the source file
      * @return mixed Translated label or FALSE on failure
      * @throws \TYPO3\Flow\I18n\TranslationProvider\Exception\InvalidPluralFormException
      */
-    public function getTranslationByOriginalLabel($originalLabel, \TYPO3\Flow\I18n\Locale $locale, $pluralForm = null, $sourceName = 'Main', $packageKey = 'TYPO3.Flow')
+    protected function getTranslationByFunction($functionName, $stringToTranslate, \TYPO3\Flow\I18n\Locale $locale = null, $quantity = null, $sourceName = 'Main', $packageKey = 'TYPO3.Flow')
     {
-        $model = $this->getModel($packageKey, $sourceName, $locale);
+        if ($locale === null) {
+            $locale = $this->localizationService->getConfiguration()->getCurrentLocale();
+        }
+
+        $translation = false;
 
-        if ($pluralForm !== null) {
-            $pluralFormsForProvidedLocale = $this->pluralsReader->getPluralForms($locale);
+        foreach ($this->localizationService->getLocaleChain($locale) as $localeInChain) {
```

Comment:
Not sure if the iteration should be so deep down in the specific TranslationProvider. It's a behaviour I would rather expect to be independent of the specific implementation.